rust/src/sysroot_upgrade.rs

```diff
@@ -7,18 +7,47 @@ use anyhow::{Context, Result};
 use std::convert::TryInto;
 use std::pin::Pin;
 
+/// `ContainerImport`'s fields are currently identical to ostree-rs-ext''s `Import`'s fields,
+/// but cxx.rs currently requires types used as extern Rust types to be defined by the same crate
+/// that contains the bridge using them, so we redefine an `ContainerImport` struct here.
+pub struct ContainerImport {
+    pub ostree_commit: String,
+    pub image_digest: String,
+}
+
+impl ContainerImport {
+    pub(crate) fn get_ostree_commit(&mut self) -> String {
```

Comment:
Do we need these accessors?  The member types are also `pub` and I think cxx should support directly reading them from the C++ side.  (We have accessors in other places because we use e.g. `Option<String>` which isn't bridged)

Reply:
ah hmm, I thought the C++ side had some trouble reading fields directly (but I had a bunch of other issues at the time as well), let me try again.

Reply:
I'm getting `error: 'struct rpmostreecxx::ContainerImport' has no member named 'ostree_commit'; did you mean 'get_ostree_commit'?` during compilation if I try to access the fields directly on the C++ side by doing e.g. `import->image_digest.c_str()`. Do I need to expose the fields in `lib.rs` similar to how we expose the functions? I wasn't able to find the documentation for this.